### ISSUE TITLE
flag moving to configmap

### ISSUE THREAD

**Issue by spyroot** [aquasecurity/kube-bench#1732]
Hi Folks,

I see more and more flag that CIS check either deprecated or moving to configmap.  Are you planning to support for checks in ConfigMap vs flag from process ?  I see many false positive where flag set in ConfigMap but not in args. 

Thank you

**Comment by afdesk** [comment]
If I understood correctly your question, Trivy can scan ConfigMap in Kubernetes cluster. Does it help you?

**Comment by spyroot** [comment]
Yes, but I was asking about kube-bench, I think some cmd can be rewritten to get the exact same value from config map vs. args vs. ps.  That will eliminate false positives. Unless, of course, kube-bench doesn't want to support new semantics.

**Comment by afdesk** [comment]
> Yes, but I was asking about kube-bench, I think some cmd can be rewritten to get the exact same value from config map vs. args vs. ps. That will eliminate false positives. Unless, of course, kube-bench doesn't want to support new semantics.

It sounds interesting. 
Could you share some samples of cmd, which can be rewritten?
thanks

**Comment by spyroot** [comment]
For example, this one if you don't have any other JSON parser.  normally you can get away with jq and jt.

Get the data from config map, parse json via jq extract clients file, for example. As follow up, since you already know a path, you can check file permission.



- id: 4.2.3
        text: "Ensure that the --client-ca-file argument is set as appropriate (Automated)"
        audit: |
          kubectl get configmap kubelet-config -n kube-system -o yaml | \
            python3 -c 'import sys, yaml, json; data = yaml.safe_load(sys.stdin); print(json.dumps(yaml.safe_load(data["data"]["kubelet"])))' | \
            jq -r ".authentication.x509.clientCAFile == \"/etc/kubernetes/pki/ca.crt\""
        audit_config: "/bin/cat $kubeletconf"
        tests:
          - test_items:
              - flag: clientCAFile
                path: '{.authentication.x509.clientCAFile}'
                value: /etc/kubernetes/pki/ca.crt
        remediation: |
          If using a Kubelet config file, edit the file to set authentication.x509.clientCAFile to
          the location of the client CA file.
          If using command line arguments, edit the kubelet service file
          $kubeletsvc on each worker node and
          set the below parameter in KUBELET_AUTHZ_ARGS variable.
          --client-ca-file=<path/to/client-ca-file>
          Based on your system, restart the kubelet service. For example,
          systemctl daemon-reload
          systemctl restart kubelet.service
        scored: true
